ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionProcessor.java

```diff
@@ -484,6 +483,9 @@ public TransactionProcessingResult processTransaction(
       final Wei coinbaseWeiDelta =
           coinbaseCalculator.price(usedGas, transactionGasPrice, blockHeader.getBaseFee());
 
+      operationTracer.traceBeforeRewardTransaction(worldUpdater, transaction, coinbaseWeiDelta);
```

Comment:
this seems unrelated

Reply:
This was a hint used by Karim to handle the miningBeneficiary case. He used the tracing feature to access the context of miningBeneficiary in ParallelizedConcurrentTransactionProcessor.runTransaction, and check if the address of the miner was touched before executing the transaction.

Reply:
Yes it is necessary for the PR and for conflict detection. I explain it in the comments normally

Reply:
In fact I use it to know if we touched the address of the beneficiary before paying the reward. If we touch it before it is a conflict if we touch it just for the reward it's ok

Reply:
Sorry for the late feedback here, but wouldn't paying the reward to the coinbase affect consensus?  If one of the parallel transactions was _from_ the coinbase, couldn't its result be affected by interim transaction rewards if the coinbase submitted a transaction that invoked BALANCE opcode?

I think we need to flatly ban coinbase from parallelization for this reason.